app/controllers/v1/ministries_controller.rb

```diff
@@ -9,6 +9,15 @@ def index
       render_ministries
     end
 
+    def update
+      ministry = Ministry.find_by(min_code: params[:id])
+      error_message = "Ministry not found for country code #{params[:id]}"
+      render status: 404, json: { error: error_message } and return if ministry.nil?
+      ministry.activate_site
+      success_message = "Ministry site activated for #{ministry.name}"
+      render status: 200, json: { success: success_message }
```

Comment:
This really should follow suit from PeopleController (and UsersController); please implement a render_ministry

Reply:
Should I leverage the `MinistrySerializer` here? It's a bit misleading as the purpose of this method is not really to fetch (or update) all of the data relevant to a ministry, it's just performing activation. Is it still best to return a full JSON representation of the updated ministry on success just in case the client wants to do anything else with it?

Reply:
That would be consistent with the other controller code, and with how update actions are supposed to behave. Otherwise, this should be a custom route, but I thought you were trying to avoid those.